lib/tasks/analyse/links.rb

```diff
@@ -167,9 +174,13 @@ def add_rows(data:, start_row_number:, worksheet:)
     end
   end
 
-  def results_as_arrays(results)
+  def results_as_arrays(results, worksheet)
     results.map do |result|
-      result_columns.map { |column| result[column] }
+      if worksheet == "Overview"
```

Comment:
I am using the `page_type` to format the columns, so I run into a problem here as the `Overview` sheet has lots of different page_types so the `section` column data gets written into `total_number_of_links_per_section` for grid and leave pages as the array of data is shorter due to mapping it depending on the columns. I know checking for string equality could be a bit brittle but in this case I think it could be ok. What do you think? Maybe we can pair on it?